### DISCUSSION TITLE
UXF PositionRotation Tracker: Rotation tracking with Euler Angles

### DISCUSSION THREAD

**Discussion by BearsBeetsBattlestarGalactica** [immersivecognition/unity-experiment-framework#157] [Q&A] [answered]
Hello,

I am having a bit of trouble again. What I am doing is tracking an object in Unity with the UXF tracker. I want to plot the direction of the object "looking" in python and therefore I am transforming the Euler Angles into normal angles. It looks like this right now:

![grafik](https://user-images.githubusercontent.com/95363686/234021341-4fe4bd9d-a65f-4fb5-adad-6c2d1ae8c81b.png)

The blue line is the trajectory of the object and the black arrows are the direction of it looking at this position

It seems like there is an offset or something else saved from unity. Do you have any ideas what the UXF trackers are tracking exactly?

Thank you very much!

**Comment by jackbrookes** [comment, marked as answer]
Hi!

UXF just logs the global rotation of the object as euler angles:

https://github.com/immersivecognition/unity-experiment-framework/blob/fbeeb775422bd250d016bfa94775707cd6b2aed7/Assets/UXF/Scripts/Trackers/PositionRotationTracker.cs#L25

Not sure exactly what you are doing to transform Euler Angles into "normal angles". There's no offset or anything like that. But there are a few things you could do:

1. Write a custom tracker script to log the "direction" of the object looking on each frame. If by direction you mean the vector of the forward direction, then you can just log `transform.forward`. If you mean something else, you could calculate it in C# then log it. This is the easiest options since you don't have to worry about different conventions. https://docs.unity3d.com/ScriptReference/Transform-eulerAngles.html
2. You could transform the euler angles back into a quaternion in Python, which may help with some calculations. Euler Angles are applied in a specific order, so you have to take that into account. In Unity, its described here:

> Euler angles can represent a three dimensional rotation by performing three separate rotations around individual axes. In Unity these rotations are performed around the Z axis, the X axis, and the Y axis, in that order. 

https://docs.unity3d.com/ScriptReference/Transform-eulerAngles.html

Unity uses a left-handed coordinate system which may also be relevant.

**Reply by BearsBeetsBattlestarGalactica** [reply]
Hello Jack,

Thanks again, 1. worked like a charme!

Keep up the great work.

Kind regards 
Dominic
